data/scripts/set_wallpaper

```diff
@@ -139,162 +112,144 @@ if [ "${KDE_FULL_SESSION}" == "true" ]; then
+    rc=$?
+    if [[ $rc = 0 ]]; then
+        for i in $(xfconf-query -c xfce4-desktop -p /backdrop -l | grep -E -e "screen.*/monitor.*image-path$" -e "screen.*/monitor.*/last-image$"); do
+            xfconf-query -c xfce4-desktop -p "$i" -n -t string -s "" 2>/dev/null
+            xfconf-query -c xfce4-desktop -p "$i" -s "" 2>/dev/null
+            xfconf-query -c xfce4-desktop -p "$i" -s "$WP" 2>/dev/null
+        done
+    fi
 
-# LXDE/PCmanFM
-if [ "$XDG_CURRENT_DESKTOP" == "LXDE" ]; then
-    pcmanfm --set-wallpaper "$WP" 2> /dev/null
-fi
+elif [ "$DE" == "lingmo" ]; then
+    # Lingmo OS
+    qdbus com.lingmo.Settings /Theme com.lingmo.Theme.setWallpaper "$WP" 2>/dev/null
 
-# LXQt/PCmanFM-qt
-if [ "$XDG_CURRENT_DESKTOP" == "LXQt" ]; then
-    pcmanfm-qt --set-wallpaper "$WP" 2> /dev/null
-fi
+elif [ "$DE" == "lxde" ]; then
+    # LXDE/PCmanFM
+    pcmanfm --set-wallpaper "$WP" 2>/dev/null
 
-# For simple WMs, use either feh or nitrogen
-# Implementation note: this uses spaces around list items to enforce matching whole words.
-# This also means that an empty variable won't cause false positives, since it expands to "  "
-SIMPLE_WMS=("bspwm" "dwm" "herbstluftwm" "i3" "i3-gnome" "i3-with-shmlog" "jwm" "jdwm" "LeftWM" "openbox" "qtile" "qtile-venv" "spectrwm" "xmonad")
-if [[ " ${SIMPLE_WMS[*]} " = *" $XDG_CURRENT_DESKTOP "* || " ${SIMPLE_WMS[*]} " = *" $XDG_SESSION_DESKTOP "* ||
-      " ${SIMPLE_WMS[*]} " = *" $DESKTOP_SESSION "* ]]; then
-    if command -v "feh" >/dev/null 2>&1; then
-        # TODO: This should take the scaling parameter $4 into account and use other options than --bg-fill
-        feh --bg-fill "$WP" 2> /dev/null
-    elif command -v "nitrogen" >/dev/null 2>&1; then
-          # TODO: This should take the scaling parameter $4 into account and use other options than --set-zoom-fill
-        nitrogen --set-zoom-fill --save "$WP" 2> /dev/null
-    fi
-fi
+elif [ "$DE" == "lxqt" ]; then
+    # LXQt/PCmanFM-qt
+    pcmanfm-qt --set-wallpaper "$WP" 2>/dev/null
 
-# Fluxbox
-if [[ "fluxbox" == "$XDG_CURRENT_DESKTOP" || "fluxbox" == "$DESKTOP_SESSION" ]]; then
-    fbsetbg "$WP" 2> /dev/null
+elif [ "$DE" == "fluxbox" ]; then
+    # Fluxbox
+    fbsetbg "$WP" 2>/dev/null
     ## fix ~/.fluxbox/lastwallpaper to using latest wallpaper when relogin fluxbox
     mkdir -p ~/.fluxbox
-    echo '$''full ''$'"full|$WP||:0.0" > ~/.fluxbox/lastwallpaper
-fi
-
-# If swaybg is available, use it as prevents system freeze.
-# See https://github.com/swaywm/sway/issues/5606
-if command -v "swaybg" >/dev/null 2>&1; then
-    # Grey background flicker is prevented by killing old swaybg process after new one.
-    # See https://github.com/swaywm/swaybg/issues/17#issuecomment-851680720
-    PID=$(pidof swaybg)
-    swaybg -i "$WP" -m fill &
-    if [ -n "$PID" ]; then
-        sleep 1
-        kill "$PID" 2>/dev/null
+    echo '$''full ''$'"full|$WP||:0.0" >~/.fluxbox/lastwallpaper
+
+elif [ "$DE" == "sway" ]; then
```

Comment:
On window managers, user can replace almost all components with whatever they like. This is why swaybg can be used independently of using sway. I use swaybg on hyprland, and this broke my setup ([hyprland is not supported by variety](https://wiki.hyprland.org/Useful-Utilities/Wallpapers/#hyprpaper), but [swaybg is](https://github.com/varietywalls/variety/pull/618/files), which works perfectly fine on hyprland)

Instead, you should check if swaybg command is available

CC @willfish @jlu5

Reply:
Even if someone would add support for hyprpaper, I like swaybg more, and want to continue using it